samples/auth_cloud_idtoken_metadata_server.rb

```diff
@@ -0,0 +1,41 @@
+# Copyright 2023 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START auth_cloud_idtoken_metadata_server]
+def auth_cloud_idtoken_metadata_server url:
+  # The url or target audience to obtain the ID token for.
+  # url = "http://www.example.com"
```

Comment:
Recommend using yardoc for the method to document this instead.

See the canonical samples in google-cloud-ruby/google-cloud-secret_manager for additional examples